src/reducers/route-reducer.ts

```diff
@@ -0,0 +1,81 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import type { AppDispatch } from "../store";
+import { startNewRoute, deactivateExistingRoute } from "../requests";
+
+interface Route {
+	routeId: string;
+	showRoute: boolean;
+	active: boolean;
+}
+
+const initialState: Route = {
+	routeId: null,
```

Comment:
We should add `strict: true` to `tsconfig`'s compiler options — `string` and `null` have no strict overlap and this is a common source for bugs! The `Route.routeId` type is clearly actually `string | null`, not `string`.

Reply:
Done